tests/shared/src/test/scala/epollcat/TcpSuite.scala

```diff
@@ -79,8 +80,15 @@ class TcpSuite extends EpollcatSuite {
 
     def accept: Resource[IO, IOSocketChannel] =
       Resource
-        .make(IO.async_[AsynchronousSocketChannel](cb => ch.accept(null, toHandler(cb))))(ch =>
-          IO(ch.close()))
+        .makeFull[IO, AsynchronousSocketChannel] { poll =>
+          poll {
+            IO.async { cb =>
+              IO(ch.accept(null, toHandler(cb)))
+                // it seems the only way to cancel accept is to close the socket :(
+                .as(Some(IO(ch.close())))
+            }
```

Comment:
The fundamental issue is that there is no good way with the JDK APIs to cancel a pending `accept` (which is really dumb to be honest). Some googling suggested that closing the channel is the only way to do so.

This is very ugly, but it is good enough for our tests ... I think.